src/app/commands/cmd_sprite_properties.cpp

```diff
@@ -40,6 +44,95 @@ namespace app {
 
+class TilesetListItem : public ui::ListItem {
+public:
+  TilesetListItem(const doc::Tileset* tileset, doc::tileset_index tsi)
+    : ListItem(app::tileset_label(tileset, tsi)),
+    m_buttons(new HBox()) {
+      m_buttons->setTransparent(true);
+      m_buttons->setExpansive(true);
+      m_buttons->setVisible(false);
+
+      auto filler = new BoxFiller();
+      filler->setTransparent(true);
+      m_buttons->addChild(filler);
+
+      auto duplicateBtn = new Button(Strings::sprite_properties_duplicate_tileset());
+      duplicateBtn->setTransparent(true);
+      duplicateBtn->Click.connect([this, tileset]{
+        auto tilesetClone = Tileset::MakeCopyCopyingImages(tileset);
+
+        Tx tx;
+        tx(new cmd::AddTileset(tileset->sprite(), tilesetClone));
+        tx.commit();
+
+        TilesetDuplicated(tilesetClone);
+      });
+      m_buttons->addChild(duplicateBtn);
+
+      auto deleteBtn = new Button(Strings::sprite_properties_delete_tileset());
+      deleteBtn->setTransparent(true);
+      deleteBtn->Click.connect([this, tileset]{
+        doc::tileset_index tsi = tileset->sprite()->tilesets()->getIndex(tileset);
+        std::string tilemapsNames;
+        for (auto layer : tileset->sprite()->allTilemaps()) {
+          auto tilemap = static_cast<doc::LayerTilemap*>(layer);
+          if (tilemap->tilesetIndex() == tsi) {
+            tilemapsNames += tilemap->name() + ", ";
+          }
+        }
+        if (!tilemapsNames.empty()) {
+          tilemapsNames = tilemapsNames.substr(0, tilemapsNames.size()-2);
+          ui::Alert::show(fmt::format(Strings::alerts_cannot_delete_used_tileset(), tilemapsNames));
+          return;
+        }
+
+        Tx tx;
+        tx(new cmd::RemoveTileset(tileset->sprite(), tsi));
+        tx.commit();
+
+        TilesetDeleted(this);
+      });
+      m_buttons->addChild(deleteBtn);
+
+      addChild(m_buttons);
+  }
+
+  bool onProcessMessage(Message* msg) override
+  {
+    switch (msg->type()) {
+      case kMouseMoveMessage:
+        auto listbox = static_cast<ListBox*>(parent());
+        if (listbox && listbox->getSelectedChild() != this) {
+          if (listbox->getSelectedChild()) {
+            static_cast<TilesetListItem*>(listbox->getSelectedChild())->m_buttons->setVisible(false);
+          }
+          listbox->selectChild(this);
+          m_buttons->setVisible(true);
+        }
+        break;
+    }
+    return ui::ListItem::onProcessMessage(msg);
+  }
+
+  void onVisible(bool visible) override
+  {
+    if (!visible) {
+      // If we don't invalidate here list items could leave old state on screen
+      // when moving the selected item.
+      // TODO: Should we add an invalidate() call on Widget::setVisible(false)? This
+      // would make this onVisible unnecessary.
+      invalidate();
+    }
+  }
+
+  obs::signal<void(TilesetListItem *)> TilesetDeleted;
+  obs::signal<void(const Tileset*)> TilesetDuplicated;
+
+private:
+  ui::HBox* m_buttons;
```

Comment:
This could be changed to:
```suggestion
  ui::HBox m_buttons;
```
Changing `m_buttons->` with `m_buttons.` and `addChild(&m_buttons);`.

Other possibility would be to create/destroy buttons on kMouseEnter/LeaveMessage (KeyItem does this). But not 100% required.